pyfftw/pyfftw.pyx

```diff
@@ -1993,15 +2006,23 @@ cdef class FFTW:
         and putting the result in the output array (i.e.
         :attr:`FFTW.output_array`).
         '''
-        cdef void *input_pointer = (
-                <void *>np.PyArray_DATA(self._input_array))
-        cdef void *output_pointer = (
-                <void *>np.PyArray_DATA(self._output_array))
-
-        cdef void *plan = self._plan
-        cdef fftw_generic_execute fftw_execute = self._fftw_execute
         with nogil:
-            fftw_execute(plan, input_pointer, output_pointer)
+            self.execute_nogil()
+
+    cdef void execute_nogil(self) noexcept nogil:
+
+        self._fftw_execute(
+            self._plan,
+            self._input_pointer,
+            self._output_pointer)
+
+cpdef void execute_in_nogil(FFTW FFTW_object):
```

Comment:
I don't fully understand the need for this function, nor for the associated test...

Reply:
Good point. 
I went back and forth a few times on how best to implement the cdef class method. I honestly though initially that those could only be accessed from within the class (private methods). This function is there merely to demonstrate that this is not the case. It adds no functionality and can be removed.